// @kkiapay-org/widget-interfaces@1.0.42/dist/lib.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListenerEventName = exports.PaymentProcess = exports.DirectDebitProcess = exports.WaveProcess = exports.BankCardsProcess = exports.MomoProcess = exports.InitProcess = exports.PaymentMethod = exports.TRANSACTION_STATUS = exports.EnvMode = void 0;
var EnvMode;
(function (EnvMode) {
    EnvMode["DEV"] = "DEV";
    EnvMode["LIVE"] = "LIVE";
    EnvMode["SANDBOX"] = "SANDBOX";
})(EnvMode = exports.EnvMode || (exports.EnvMode = {}));
var TRANSACTION_STATUS;
(function (TRANSACTION_STATUS) {
    TRANSACTION_STATUS["PENDING"] = "PENDING";
    TRANSACTION_STATUS["SUCCESS"] = "SUCCESS";
    TRANSACTION_STATUS["FAILED"] = "FAILED";
    TRANSACTION_STATUS["INIT"] = "INIT";
})(TRANSACTION_STATUS = exports.TRANSACTION_STATUS || (exports.TRANSACTION_STATUS = {}));
var PaymentMethod;
(function (PaymentMethod) {
    PaymentMethod["momo"] = "momo";
    PaymentMethod["card"] = "card";
    PaymentMethod["direct_debit"] = "direct_debit";
})(PaymentMethod = exports.PaymentMethod || (exports.PaymentMethod = {}));
var InitProcess;
(function (InitProcess) {
    InitProcess["user_information"] = "user_information";
})(InitProcess = exports.InitProcess || (exports.InitProcess = {}));
var MomoProcess;
(function (MomoProcess) {
    MomoProcess["waiting_user_feedback"] = "waiting_user_feedback";
    MomoProcess["code_pin"] = "code_pin";
})(MomoProcess = exports.MomoProcess || (exports.MomoProcess = {}));
var BankCardsProcess;
(function (BankCardsProcess) {
    BankCardsProcess["need_user_pin"] = "need_user_pin";
    BankCardsProcess["need_otp_verification"] = "need_otp_verification";
    BankCardsProcess["need_billing_details"] = "need_billing_details";
    BankCardsProcess["need_link_verification"] = "need_link_verification";
})(BankCardsProcess = exports.BankCardsProcess || (exports.BankCardsProcess = {}));
var WaveProcess;
(function (WaveProcess) {
    WaveProcess["wave_launch_url"] = "wave_launch_url";
})(WaveProcess = exports.WaveProcess || (exports.WaveProcess = {}));
var DirectDebitProcess;
(function (DirectDebitProcess) {
    DirectDebitProcess["enrollment"] = "enrollment";
    DirectDebitProcess["confirm_phone"] = "confirm_phone";
    DirectDebitProcess["confirm_payment"] = "confirm_payment";
})(DirectDebitProcess = exports.DirectDebitProcess || (exports.DirectDebitProcess = {}));
var PaymentProcess;
(function (PaymentProcess) {
    PaymentProcess["user_information"] = "user_information";
    PaymentProcess["waiting_user_feedback"] = "waiting_user_feedback";
    PaymentProcess["code_pin"] = "code_pin";
    PaymentProcess["need_user_pin"] = "need_user_pin";
    PaymentProcess["need_otp_verification"] = "need_otp_verification";
    PaymentProcess["need_billing_details"] = "need_billing_details";
    PaymentProcess["need_link_verification"] = "need_link_verification";
    PaymentProcess["wave_launch_url"] = "wave_launch_url";
    PaymentProcess["enrollment"] = "enrollment";
    PaymentProcess["confirm_phone"] = "confirm_phone";
    PaymentProcess["confirm_payment"] = "confirm_payment";
})(PaymentProcess = exports.PaymentProcess || (exports.PaymentProcess = {}));
var ListenerEventName;
(function (ListenerEventName) {
    ListenerEventName["NETWORK_STATE_CHANGED"] = "NETWORK_STATE_CHANGED";
    ListenerEventName["INIT_WIDGET"] = "INIT_WIDGET";
    ListenerEventName["WIDGET_SUCCESSFULLY_INIT"] = "WIDGET_SUCCESSFULLY_INIT";
    ListenerEventName["CLOSE_WIDGET"] = "CLOSE_WIDGET";
    ListenerEventName["DESTROY_WIDGET"] = "DESTROY_WIDGET";
    ListenerEventName["WIDGET_SUCCESSFULLY_DESTROYED"] = "WIDGET_SUCCESSFULLY_DESTROYED";
    ListenerEventName["PAYMENT_INIT"] = "PAYMENT_INIT";
    ListenerEventName["PAYMENT_ABORTED"] = "PAYMENT_ABORTED";
    ListenerEventName["PENDING_PAYMENT"] = "PENDING_PAYMENT";
    ListenerEventName["ON_USER_FEEDBACK"] = "ON_USER_FEEDBACK";
    ListenerEventName["PAYMENT_FAILED"] = "PAYMENT_FAILED";
    ListenerEventName["PAYMENT_SUCCESS"] = "PAYMENT_SUCCESS";
    ListenerEventName["PAYMENT_END"] = "PAYMENT_END";
    ListenerEventName["RETRY_PAYMENT"] = "RETRY_PAYMENT";
    ListenerEventName["WAVE_LINK"] = "WAVE_LINK";
})(ListenerEventName = exports.ListenerEventName || (exports.ListenerEventName = {}));
